rest/route/stats.go

```diff
@@ -453,6 +453,19 @@ func (tsh *taskStatsHandler) Run(ctx context.Context) gimlet.Responder {
 	if err != nil {
 		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "Failed to retrieve the task stats"))
 	}
+	if len(taskStatsResult) == 0 {
+		// Lookup last sync date for project
+		statsStatus, err := stats.GetStatsStatus(tsh.filter.Project)
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "Failed to retrieve stats status"))
+		}
+		if time.Since(statsStatus.LastJobRun).Hours() > time.Duration.Hours(24) {
```

Comment:
It seems like this assumes that the user is asking for the latest stats and not a specific date range. How is it supposed to behave if they specify a date range in the past and it's not available?

Reply:
changed to checking the "afterdate" 
we always require before and after fields